platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +180,28 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
+                IOUtils.copy(is, tfbos);
+                try (InputStream inputStream = tfbos.asByteSource()
+                        .openStream()) {
+                    Parser parser = new AutoDetectParser();
+                    ToXMLContentHandler xmlContentHandler = new ToXMLContentHandler();
+                    Metadata metadata = new Metadata();
+                    parser.parse(inputStream, xmlContentHandler, metadata, new ParseContext());
+
+                    String metacardContentType = metadata.get(Metadata.CONTENT_TYPE);
+                    fileExtension = getFileExtensionForMimeType(metacardContentType).replace(".",
+                            "");
+                }
+                is = tfbos.asByteSource()
```

Comment:
I think this should be in a `finally`. That way if anything inside this `try` errors out, then we'll still have an input stream to use for the rest of the method.